nixos/modules/services/misc/home-assistant.nix

```diff
@@ -105,20 +130,21 @@ in {
     systemd.services.home-assistant = {
       description = "Home Assistant";
       after = [ "network.target" ];
-      preStart = lib.optionalString (cfg.config != null) ''
-        rm -f ${cfg.configDir}/configuration.yaml
-        ln -s ${configFile} ${cfg.configDir}/configuration.yaml
+      preStart = lib.optionalString (cfg.applyDefaultConfiguration || cfg.config != null) ''
```

Comment:
That means that the user is asking to apply the default configuration but it doesn't actually do anything unless config is set. That seems less obvious. Or are you concerned about people having their manually created configuration.yaml overwritten?

Reply:
Since the default for `applyDefaultConfiguration` (which could be shortened to `applyDefaultConfig`) is `true`, the user is not really asking for it.
I'm especially thinking about users currently running Home Assistant and not using the `config` option. They would indeed have their config overwritten when updating their system.
We could
- set the default for `applyDefaultConfig` to `false` or
- leave it `true`, change above line to only use `cfg.config == null` and maybe document that `applyDefaultConfig` only has an effect if `config != null`.